### PR TITLE
Closes #2348 - adds feature to request the current user via /users?currentUser

### LINKED ISSUE #2348: Adds query for currest user data via users resource
Enable to get the data of the current user via the following:
GET /api/v1/users?current-user

### PR DESCRIPTION
**Pull request by nyuuyn**
See issue [#2348](https://github.com/Taskana/taskana/issues/2348), [sonar](https://sonarcloud.io/summary/new_code?id=nyuuyn_taskana&branch=TSK-2348)
---
Release Notes:

```
Enables fetching the current user data via HTTP GET on /users?currentUser
```
 
### For the submitter:
- [ ] I updated the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) and will supply links to the specific files
- [ ] I did not update the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview)
- [x] I included a link to the [SonarCloud branch analysis](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1019969636/SonarCloud+Integration)
- [x] After integration of the PR, I added a description of changes on the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [ ] I did not update the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [ ] I put the ticket in review
- [ ] After integration of the pull request, I verified our [bluemix test environment](http://taskana.mybluemix.net/taskana) is not broken

### Verified by the reviewer:
- [ ] Commit message format → (Closes) #<Issue Number>: Your commit message. i.e. Closes #2271: Your commit message.
- [ ] Submitter's update to [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) is sufficient
- [ ] SonarCloud analysis meets our standards
- [ ] Update of the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana) reflects changes
- [ ] PR fulfills the ticket
- [ ] Edge cases and unwanted side effects are tested
- [ ] Readability

### CODE REVIEW

File: rest/taskana-rest-spring/src/main/java/pro/taskana/user/rest/UserController.java [hunk 1]
```diff
@@ -54,24 +61,36 @@ public class UserController {
...
   }
 
   /**
    * This endpoint retrieves multiple Users. If a userId can't be found in the database it will be
    * ignored. If none of the given userIds is valid, the returned list will be empty.
+   * If currentUser is set, the current User from the context will be retrieved as well
    *
    * @title Get multiple Users
    * @param userIds the ids of the requested Users
+   * @param currentUser boolean to indicate whether to fetch the current user or not
```
Comment (ryzheboka) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/user/rest/UserController.java:74]: The param "currentUser" is a boolean only logically but not from the implementation standpoint. What do you think about changing the description here? It should be clear which values for currentUser are valid

**Review by ryzheboka — CHANGES_REQUESTED** [review]

File: rest/taskana-rest-spring/src/test/java/pro/taskana/user/rest/UserControllerIntTest.java [hunk 2]
```diff
@@ -63,6 +63,42 @@ void should_ReturnExistingUsers() throws Exception {
         .containsExactlyInAnyOrder("Max", "Elena");
   }
 
+  @Test
+  void testGetCurrentUser() {
```
[the remark this answers is not in the document]
Comment (nyuuyn) [reply rest/taskana-rest-spring/src/test/java/pro/taskana/user/rest/UserControllerIntTest.java]: fixed

File: rest/taskana-rest-spring/src/main/java/pro/taskana/user/rest/UserController.java (hunk 1 above)
Comment (nyuuyn) [reply rest/taskana-rest-spring/src/main/java/pro/taskana/user/rest/UserController.java]: fixed

File: rest/taskana-rest-spring/src/test/java/pro/taskana/user/rest/UserControllerIntTest.java [hunk 3]
```diff
@@ -63,6 +63,95 @@ void should_ReturnExistingUsers() throws Exception {
...
+                    HttpMethod.GET,
+                    auth,
+                    ParameterizedTypeReference.forType(UserCollectionRepresentationModel.class));
+    assertThat(response.getBody()).isNotNull();
+    assertThat(response.getBody().getContent()).hasSize(1);
+    assertThat(response.getBody().getContent()).extracting("userId").containsExactly("teamlead-1");
+  }
+
+  @Test
+  void should_ReturnCurrentUserWithValue() {
```
Comment (jamesrdi) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/user/rest/UserControllerIntTest.java:83]: Why do you need this test, the same thing is tested in should_ReturnExceptionCurrentUserWithBadValue

Comment (nyuuyn) [reply rest/taskana-rest-spring/src/test/java/pro/taskana/user/rest/UserControllerIntTest.java]: fixed

**Review by ryzheboka — APPROVED** [review]

**Pull request merged**